src/main/java/com/example/raiseTechcoursetask10/controller/SkiresortController.java

```diff
@@ -52,17 +54,26 @@ public ResponseEntity<Map<String, String>> noResourceFound(
         return new ResponseEntity(body, HttpStatus.NOT_FOUND);
     }
 
-
     @PostMapping("/skiresorts")
-    public ResponseEntity<SkiresortResponse> createSkiresort(@RequestBody SkiresortCreateForm skiresortCreateForm) {
+    public ResponseEntity<SkiresortResponse> createSkiresort(@RequestBody SkiresortCreateForm skiresortCreateForm, HttpServletRequest request) {
         Skiresort skiresort = skiresortService.createSkiresort(skiresortCreateForm);
 
-        // skiresortを作成する処理
+        // skiresortオブジェクトを元にレスポンス用のオブジェクトを生成
         SkiresortResponse skiresortResponse = new SkiresortResponse(skiresort);
-        URI url = UriComponentsBuilder.fromUriString("http://localhost8080")
+        // HttpServletRequestのインスタンスでリクエストの中身を取得し、動的なURLを生成
+        URI url = UriComponentsBuilder.fromUriString(request.getRequestURI())
                 .path("/skiresorts/{id}")
                 .buildAndExpand(skiresort.getId())
                 .toUri();
         return ResponseEntity.created(url).body(skiresortResponse);
     }
+
+    @PatchMapping("/skiresorts/{id}")
+    public ResponseEntity<Map<String, String>> update(@PathVariable("id") int id, @RequestBody @Validated SkiresortCreateForm form) {
+
+        SkiresortUpdateForm skiresortUpdateForm = new SkiresortUpdateForm(id, form.getName(), form.getArea(), form.getCustomerEvaluation());
+        // SkiresortUpdateFormの情報を使用してレコードを更新し、更新されたSkiresortエンティティを返す
+        Skiresort updatedSkiresort = skiresortService.updateSkiresort(new SkiresortUpdateForm(id, form.getName(), form.getArea(), form.getCustomerEvaluation()));
```

Comment:
登録と違い更新は戻り値なくても良さそう。
```suggestion
        skiresortService.updateSkiresort(new SkiresortUpdateForm(id, form.getName(), form.getArea(), form.getCustomerEvaluation()));
```

Reply:
SkiresortServiceImplで戻り値なし（void）に変更したので、Controllerも修正します。

Reply:
修正漏れでした。